client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/configure-sms-notification/phone-number-editor.tsx

```diff
@@ -1,22 +1,131 @@
+import FormPhoneInput from 'calypso/components/forms/form-phone-input';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import { useSelector } from 'calypso/state';
+import getCountries from 'calypso/state/selectors/get-countries';
+import type { StateMonitorSettingsSMS } from '../../sites-overview/types';
 
 interface Props {
 	toggleModal: () => void;
+	allPhoneItems: Array< StateMonitorSettingsSMS >;
+	setAllPhoneItems: ( phoneNumbers: Array< StateMonitorSettingsSMS > ) => void;
 }
 
-export default function PhoneNumberEditor( { toggleModal }: Props ) {
+export default function PhoneNumberEditor( {
+	toggleModal,
+	allPhoneItems,
+	setAllPhoneItems,
+}: Props ) {
 	const translate = useTranslate();
 
+	const countriesList = useSelector( ( state ) => getCountries( state, 'sms' ) ?? [] );
+
+	const [ showCodeVerification, setShowCodeVerification ] = useState< boolean >( false );
+	const [ validationStatus, setValidationStatus ] = useState< {
+		isValid: boolean;
+		errorMessage: string;
+	} >( {
+		isValid: true,
+		errorMessage: '',
+	} );
+	const [ validationError, setValidationError ] = useState<
+		{ phone?: string; code?: string } | undefined
+	>();
+	const [ smsItem, setSMSItem ] = useState< {
+		name: string;
+		countryCode: string;
+		phoneNumber: string;
+		phoneNumberFull: string;
+	} >( {
+		name: '',
+		countryCode: '',
+		phoneNumber: '',
+		phoneNumberFull: '',
+	} );
+
+	const handleSetSMSItems = useCallback(
+		( isVerified = true ) => {
+			const updatedPhoneItem = {
+				...smsItem,
+				verified: isVerified,
+			};
+			// Check if exists when editing
+			allPhoneItems.push( updatedPhoneItem );
+			setAllPhoneItems( allPhoneItems );
+			toggleModal();
+		},
+		[ allPhoneItems, smsItem, setAllPhoneItems, toggleModal ]
+	);
+
+	const handleRequestVerificationCode = () => {
+		setShowCodeVerification( true );
+		// TODO: Make API call to request verification code
+	};
+
+	// Save unverified phone number to the list when user clicks on Later button
+	function onSaveLater() {
+		handleSetSMSItems( false );
+	}
+
 	const onSave = ( event: React.FormEvent< HTMLFormElement > ) => {
 		event.preventDefault();
-
-		// Handle save here
+		setValidationError( undefined );
+		if ( validationStatus.isValid ) {
+			if ( showCodeVerification ) {
+				// TODO: Add code verification
+			}
+			return handleRequestVerificationCode();
+		}
+		setValidationError( { phone: validationStatus.errorMessage } );
 	};
 
+	const handleChange = useCallback(
+		( key ) => ( event: React.ChangeEvent< HTMLInputElement > ) => {
+			setSMSItem( ( prevState ) => ( { ...prevState, [ key ]: event.target.value } ) );
+		},
+		[]
+	);
+
 	const title = translate( 'Add your phone number' );
 	const subTitle = translate( 'Please use an accessible phone number. Only alerts sent.' );
 
+	const onChangePhoneInput = ( {
+		phoneNumberFull,
+		phoneNumber,
+		countryData,
+		isValid,
+		validation,
+	}: {
+		phoneNumberFull: string;
```

Comment:
Not a blocker, but since this anonymous type has enough fields on it. I would suggest adding an alias like `FormPhoneInputChangeResult` for readability.

Reply:
Totally agree. I have made the changes - https://github.com/Automattic/wp-calypso/pull/78000/commits/6d43ded9f22ca5f25d88ad93a6ce94410556cec2